configure.ac

```diff
@@ -388,6 +391,31 @@ case "$host_os" in
 	;;
 esac
 
+while : ; do
+	case ${file_sync} in
+	fdatasync)
+		if test x$HAVE_FDATASYNC != xyes; then
+			break
+		fi
+		file_sync=fsync;;
+	fsync)
```

Comment:
There were other issues (fixed not) but I'd say the iterated fallback
scheme is quite recognizable.